### PR TITLE
Add shellcheck

### PR DESCRIPTION
**Pull request by kierenevans**
* Add [shellcheck](https://github.com/koalaman/shellcheck) to Travis to scan shell scripts for potentially dangerous issues and general good practises.
* Fix up problems found.

### CODE REVIEW

**Comment by sroze** [comment]
Using Travis.org instead of .com, closing and opening to trigger a build.

File: .travis.yml [hunk 1]
```diff
@@ -38,3 +39,6 @@ script:
     - bin/console doctrine:migrations:migrate --no-interaction
     - vendor/bin/behat --profile=smoke -fprogress
     - popd
+
+    # Shell script tests
+    - for file in $(find . -type f \( -perm +111 -or -name "*.sh" \) ! -path "*/vendor/*" ! -path "*/.git/*" ! -path "*/api/bin/console" ! -path "*/api/bin/symfony_requirements" | sed 's#^\.#/app#'); do docker run -v "$(pwd):/app" koalaman/shellcheck:v0.4.7 --exclude SC1091 "$file"; done
```
Comment (sroze) [inline .travis.yml]: As this is fast, could you move it after the PHPUnit tests?

File: api/docker/usr/local/share/cron/run-every-x-seconds.sh [hunk 2]
```diff
@@ -7,18 +7,19 @@ error="no"
 
 if [ 'x'"$runCommand" != 'x' ]
 then
-    if [ 'x'$inputPeriod != 'x' ]
+    if [ 'x'"$inputPeriod" != 'x' ]
     then
-        loops=$(( $RUN_TIME / $inputPeriod ))
+        loops=$(( RUN_TIME / inputPeriod ))
```
Comment (sroze) [inline api/docker/usr/local/share/cron/run-every-x-seconds.sh]: Really, we don't need the `$`s here?!

Comment (kierenevans) [reply api/docker/usr/local/share/cron/run-every-x-seconds.sh]: Seems to work both ways:
```bash
bash-3.2$ export RUN_TIME=10
bash-3.2$ export inputPeriod=2
bash-3.2$ loops=$(( RUN_TIME / inputPeriod ))
bash-3.2$ echo $loops
5
bash-3.2$ loops=$(( $RUN_TIME / $inputPeriod ))
bash-3.2$ echo $loops
5
```
https://github.com/koalaman/shellcheck/wiki/SC2004 has the explanation why they recommend this change

File: .travis.yml (hunk 1 above)
Comment (kierenevans) [reply .travis.yml]: Sorted - also tried to ensure docker is available as the build failed

**Comment by sroze** [comment]
Travis is definitely unhappy about your changes :)

**Comment by sroze** [comment]
Youp youp! Thank you Kieren! Can you squash your commits?

**Comment by kierenevans** [comment]
Squashed to 3 but I can squash further if wanted :)
That was a mission!

**Comment by sroze** [comment]
Yeah, I've seen that 😅

**Pull request merged**